lib/trento/software_updates/discovery.ex

```diff
@@ -39,11 +39,13 @@ defmodule Trento.SoftwareUpdates.Discovery do
 
   @spec discover_software_updates :: {:ok, {list(), list()}}
   def discover_software_updates do
+    authenticated = setup()
```

Comment:
Can we keep this encapsulated in auth?

I was imagining that the auth either returns a previous authentication or authenticates if none previously available.

Besides that suma module would re-issue auth when a previously available one starts raising auth errors (read refresh cookie)

Reply:
This was actually intentional. I want to give the client (`discovery.ex` in this case) the option to use the auth module as it wants. If it wants to authenticate, so be it.
At the end, in this "batch" discovery, if the given authentication values are wrong, all the hosts will try to authenticate otherwise. With this approach, we try an initial authentication, and if it fails, we basically don't try anymore, as all of the hosts will fail authenticating.

This is a best-effort approach. If for some circunstance, we have an expired cookie, this auth will return it, but the 1st host will try to authenticate, it will fail, and retry (just blocking first this 1st host).

It just a matter of not doing operations when we know that authentication will fail in N hosts.

> I was imagining that the auth either returns a previous authentication or authenticates if none previously available.

Indeed it does.

Reply:
Correct, I looked closer and indeed it does what it is supposed to do. And since it was intentional, I am ok with that. Well done!

I guess I can live with the extra `Discovery.discover_host_software_updates/3` differing from the other with arity 2. Even tho it makes my eye tick a bit I get that the the difference is in the fact that we might want to provide that function a pre-authenticated state.

Can we at least change `authenticated = setup()` to `authentication = setup()`? OCD eheh